src/common/SurgePatch.cpp

```diff
@@ -1752,11 +1752,15 @@ void SurgePatch::load_xml(const void* data, int datasize, bool is_preset)
       TiXmlElement* mw = TINYXML_SAFE_TO_ELEMENT(patch->FirstChild("modwheel"));
       if (mw)
       {
-        // TODO: FIX SCENE ASSUMPTION
-         if (mw->QueryDoubleAttribute("s0", &d) == TIXML_SUCCESS)
-            ((ControllerModulationSource*)scene[0].modsources[ms_modwheel])->set_target(d);
-         if (mw->QueryDoubleAttribute("s1", &d) == TIXML_SUCCESS)
-            ((ControllerModulationSource*)scene[1].modsources[ms_modwheel])->set_target(d);
+         for (int sc = 0; sc < n_scenes; sc++)
+         {
+            char str[32];
+            sprintf(str, "s%d", sc);
```

Comment:
It doesn't matter here, but generally `snprintf( str, 32 "3%d", sc )` is safer than sprintf since it force avoids overflows.